dev_guide/builds.adoc

```diff
@@ -17,6 +17,68 @@ runnable images to be used on OpenShift. There are three types of builds:
 - STI build
 - custom build
 
+== Defining an incremental buildConfig
+STI builds provide an option to perform incremental build, which means they will
+reuse artifacts from previously built image. To create an incremental build one
+needs to create following BuildConfig:
+
+[incremental.json]
+----
+{
+  "kind": "BuildConfig",
+    "apiVersion": "v1",
+    "metadata": {
+      "name": "ruby-incremental-build",
+    },
+  "spec": {
+    "triggers": [
+      {
+        "type": "github",
+        "github": {
+          "secret": "secret101"
+        }
+      },
+      {
+        "type": "generic",
+        "generic": {
+          "secret": "secret101"
+        }
+      },
+      {
+        "type": "ImageChange"
+      }
+    ],
+    "source": {
+      "type": "Git",
+      "git": {
+        "uri": "git://github.com/openshift/ruby-hello-world.git"
+      }
+    },
+    "strategy": {
+      "type": "Source",
+      "sourceStrategy": {
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "incremental-image <1>"
+        },
+        "incremental": true <2>
+      }
+    },
+    "output": {
+      "to": {
+        "kind": "ImageStreamTag",
+        "name": "origin-incremental-sample:latest"
+      }
+    },
+  },
+}
+----
+
+<1> You need to specify here an image that supports incremental builds. Unfortunately none of the images provided by OpenShift qualify as such.
+<2> Here's how you force the builds to perform incremental build. If you specify this flag to non-incremental builder, the build will work, but you'll get an information in the logs that the incremental build wasn't successful because of missing `save-artifacts` script.
```

Comment:
In case of our images - it will, because for images where scripts are inside the image we'll try to run it and fail with a warning (see [here](https://github.com/openshift/source-to-image/blob/master/pkg/build/strategies/sti/sti.go#L126)). But for downloadable scripts we'll just put an info only on loglevel=1 (see [here](https://github.com/openshift/source-to-image/blob/master/pkg/build/strategies/sti/sti.go#L120)). Do you think we should always show it in the 2nd case?